packages/ckeditor5-dev-build-tools/src/plugins/replace.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import { walk, type Node } from 'estree-walker';
+import type { ImportDeclaration, ExportAllDeclaration, ExportNamedDeclaration } from 'estree';
+import MagicString from 'magic-string';
+import type { Plugin } from 'rollup';
+
+export interface RollupReplaceOptions {
+
+	/**
+	 * Array containing tuples of pattern and replace value. RegExp must have the `/g` flag.
+	 *
+	 * @example
+	 * [
+	 *   [ 'find', 'replace' ],
+	 *   [ /find/g, 'replace' ]
+	 * ]
+	 *
+	 * @default []
+	 */
+	replace: Array<[ RegExp | string, string ]>;
+}
+
+export function replaceImports( pluginOptions: RollupReplaceOptions ): Plugin {
+	const options: Required<RollupReplaceOptions> = Object.assign(
+		{ replace: [] },
+		pluginOptions
+	);
+
+	function isModule( node: Node ): node is ImportDeclaration | ExportAllDeclaration | ExportNamedDeclaration {
+		return [
+			'ImportDeclaration',
+			'ExportAllDeclaration',
+			'ExportNamedDeclaration'
+		].includes( node.type );
+	}
```

Comment:
Do these three types of nodes cover all cases? I see in [the docs](https://doc.esdoc.org/github.com/mason-lang/esast/identifiers.html) that there are also some additional types:
- `ExportDefaultDeclaration`,
- `ExportSpecifier`,
- `ImportDefaultSpecifier`,
- `ImportNamespaceSpecifier`,
- `ImportSpecifier`,
- `ImportSpecifierAbstract`.

Reply:
#### `ExportDefaultDeclaration`

`ExportDefaultDeclaration` cannot have any path, so there's nothing to check. Example:

```js
export default 123;
^^^^^^^^^^^^^^^^^^

// This is not possible
export default from 'test';

// This is `ExportNamedDeclaration`
export { default as default } from 'test';
```

#### `ExportSpecifier`

`ExportSpecifier` represents values inside the `export {}`. Example:

```js
export { test as test } from 'test';
         ^^^^^^^^^^^^
```

#### `ImportDefaultSpecifier`

`ImportDefaultSpecifier` represents name for the `default` export of another module.

```js
import Test from 'test';
       ^^^^
```

#### `ImportNamespaceSpecifier`

`ImportNamespaceSpecifier` represents name for the namespaces export of another module.

```js
import * as Test from 'test';
       ^^^^^^^^^
```

#### `ImportSpecifier`

`ExportSpecifier` represents values inside the `import {}`. Example:

```js
import { test as test } from 'test';
         ^^^^^^^^^^^^
```

#### `ImportSpecifierAbstract`

No idea

---

Source: https://astexplorer.net/